src/main/groovy/com/morpheus/scvmm/sync/IpPoolsSync.groovy

```diff
@@ -0,0 +1,307 @@
+                                range.externalId = masterItem.ID
+                                morpheusContext.async.cloud.network.pool.poolRange.save(range).blockingGet()
+                            }
+                        }
+                    }
+
+                    // Update the pool (if needed)
+                    def doSave = false
+                    if(existingItem.name != masterItem.Name) {
+                        existingItem.name = masterItem.Name
+                        doSave = true
+                    }
+
+                    def displayName = "${masterItem.Name} (${masterItem.Subnet})"
+                    if(existingItem.displayName != displayName) {
+                        existingItem.displayName = displayName
+                        doSave = true
+                    }
+
+                    if(existingItem.ipCount != (masterItem.TotalAddresses ?: 0).toInteger()) {
+                        existingItem.ipCount = (masterItem.TotalAddresses ?: 0).toInteger()
+                        doSave = true
+                    }
+
+                    if(existingItem.ipFreeCount != (masterItem.AvailableAddresses ?: 0).toInteger()) {
+                        existingItem.ipFreeCount = (masterItem.AvailableAddresses ?: 0).toInteger()
+                        doSave = true
+                    }
+
+                    def gateway = masterItem.DefaultGateways ? masterItem.DefaultGateways.first() : null
+                    if(existingItem.gateway != gateway) {
+                        existingItem.gateway = gateway
+                        doSave = true
+                    }
+
+                    def info = new SubnetUtils(masterItem.Subnet).info
+                    if(existingItem.netmask != info.netmask) {
+                        existingItem.netmask = info.netmask
+                        doSave = true
+                    }
+
+                    if(existingItem.subnetAddress != info.networkAddress) {
+                        existingItem.subnetAddress = info.networkAddress
+                        doSave = true
+                    }
+
+                    if (doSave == true) {
+                        morpheusContext.async.cloud.network.pool.save(existingItem).blockingGet()
+                    }
+
+                    def existingPermission = morpheusContext.services.resourcePermission.find(new DataQuery()
+                            .withFilter('morpheusResourceType', 'NetworkPool')
+                            .withFilter('morpheusResourceId', existingItem.id)
+                            .withFilter('account', cloud.account))
+                    if(!existingPermission) {
+                        def resourcePerm = new ResourcePermission(morpheusResourceType:'NetworkPool', morpheusResourceId:existingItem.id, account:cloud.account)
+                        morpheusContext.async.resourcePermission.save(resourcePerm).blockingGet()
```

Comment:
shouldn't this be a create and not a save?

Reply:
@ddeyoung changes has been done as suggested